v2.0.0 fsverity test failure on s390x (Fedora rawhide and F41)

### Description

Building containerd v2.0.0 for Fedora on supported architectures. I am seeing the following results in the build log on s390x machines:


```
?   	github.com/containerd/containerd/v2/internal/truncindex	[no test files]
ok  	github.com/containerd/containerd/v2/internal/failpoint	1.004s
--- FAIL: TestIsEnabled (0.00s)
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x3036de]
goroutine 8 [running]:
testing.tRunner.func1.2({0x3486c0, 0x5f2e70})
	/usr/lib/golang/src/testing/testing.go:1632 +0x2c4
testing.tRunner.func1()
	/usr/lib/golang/src/testing/testing.go:1635 +0x3dc
panic({0x3486c0, 0x5f2e70})
	/usr/lib/golang/src/runtime/panic.go:785 +0x11a
github.com/containerd/containerd/v2/internal/fsverity.TestIsEnabled(0xc0000ecb60)
	/builddir/build/BUILD/containerd-2.0.0-build/containerd-2.0.0/internal/fsverity/fsverity_test.go:127 +0x4ce
testing.tRunner(0xc0000ecb60, 0x3d0940)
	/usr/lib/golang/src/testing/testing.go:1690 +0x132
created by testing.(*T).Run in goroutine 1
	/usr/lib/golang/src/testing/testing.go:1743 +0x46e
FAIL	github.com/containerd/containerd/v2/internal/fsverity	0.006s
ok  	github.com/containerd/containerd/v2/internal/kmutex	0.206s
ok
```

Any insights on possible causes much appreciated. 


### Steps to reproduce the issue

1. Build containerd via Fedora spec file on s390x vm
2.
3.


### Describe the results you received and expected

The log file extract above describes the results I am seeing.

I expected the build and test suite to succeed as it does on other supported platforms.

### What version of containerd are you using?

v2.0.0

### Any other relevant information

The test suite is called with:

```
+ /usr/bin/make -O -j3 V=1 VERBOSE=1 DATADIR=/usr/share DESTDIR=/builddir/build/BUILD/containerd-2.0.0-build/BUILDROOT 'EXTRA_LDFLAGS=-Wl,-z,relro -Wl,--as-needed   -Wl,-z,now -specs=/usr/lib/rpm/redhat/redhat-hardened-ld -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1  -Wl,--build-id=sha1 -specs=/usr/lib/rpm/redhat/redhat-package-notes  -X github.com/containerd/containerd/version=2.0.0 -B 0x6ac22d4d0f899157fa535349847cb75be441746d -compressdwarf=false -linkmode=external -extldflags '\''-Wl,-z,relro -Wl,--as-needed   -Wl,-z,now -specs=/usr/lib/rpm/redhat/redhat-hardened-ld -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1  -Wl,--build-id=sha1 -specs=/usr/lib/rpm/redhat/redhat-package-notes  '\''' 'GO_BUILD_FLAGS=-buildmode pie -compiler gc -tags="rpm_crashtraceback " -a -v -x' MANDIR=/usr/share/man PREFIX=/usr REVISION=1.fc42 SHIM_CGO_ENABLED=1 VERSION=2.0.0 test
+ test
```

### Show configuration if it is related to CRI plugin.

_No response_

--- a/internal/fsverity/fsverity_test.go
+++ b/internal/fsverity/fsverity_test.go
@@ -29,6 +29,7 @@ import (
 	"testing"
 
 	"github.com/containerd/containerd/v2/pkg/testutil"
+	"golang.org/x/sys/unix"
 )
 
 type superblockROFeatures struct {
@@ -143,8 +144,7 @@ func resolveDevicePath(path string) (_ string, e error) {
 	}
 
 	// resolve to device path
-	maj := (stat.Dev >> 8) & 0xff
-	min := stat.Dev & 0xff
+	major, minor := unix.Major(stat.Dev), unix.Minor(stat.Dev)
 
 	m, err := os.Open("/proc/self/mountinfo")
 	if err != nil {
@@ -162,7 +162,7 @@ func resolveDevicePath(path string) (_ string, e error) {
 	scanner := bufio.NewScanner(m)
 
 	var entry string
-	sub := fmt.Sprintf("%d:%d", maj, min)
+	sub := fmt.Sprintf("%d:%d", major, minor)
 	for scanner.Scan() {
 		if strings.Contains(scanner.Text(), sub) {
 			entry = scanner.Text()

--- a/internal/fsverity/fsverity_test.go
+++ b/internal/fsverity/fsverity_test.go
@@ -123,8 +123,8 @@ func TestIsEnabled(t *testing.T) {
 		t.Errorf("fsverity Enable failed: %s", err.Error())
 	}
 
-	if enabled, err := IsEnabled(verityFile); !enabled || err != nil {
-		t.Errorf("expected fsverity to be enabled on file, received enabled: %t; error: %s", enabled, err.Error())
+	if enabled, err := IsEnabled(verityFile); !enabled && err != nil {
+		t.Errorf("expected fsverity to be enabled on file, received enabled: %t; error: %v", enabled, err)
 	}
 }
 

